chore: update configs for gen2 (#5211)
aws-amplify/amplify-ui: docs/src/pages/[platform]/connected-components/authenticator/configuration/loginMechanisms.react-native.mdx

</Alert>

By default, the Authenticator creates new users in the Amazon Cognito UserPool based on a unique `username`, `email` or `phone_number`.

> **Note**: A `username`, `email`, or `phone_number` value is required for Cognito User Pools.

> <Tabs.Container defaultValue="Email">

Author: Made email the default example, since username isn't fully supported

  <Tabs.List>
    <Tabs.Item value="Email">Email</Tabs.Item>
    <Tabs.Item value="Phone Number">Phone Number</Tabs.Item>
    <Tabs.Item value="Username">Username</Tabs.Item>
  </Tabs.List>
  <Tabs.Panel value="Email">
    ```jsx expoSnack
    <Authenticator.Provider>
      <Authenticator loginMechanisms={['email']}>
        <SignOutButton />
      </Authenticator>
    </Authenticator.Provider>
    ```
  </Tabs.Panel>
  <Tabs.Panel value="Phone Number">
    ```jsx expoSnack
    <Authenticator.Provider>
      <Authenticator loginMechanisms={['phone_number']}>
        <SignOutButton />
      </Authenticator>
    </Authenticator.Provider>
    ```
  </Tabs.Panel>
    <Tabs.Panel value="Username">

>     Note: Login with Username is not currently supported using Amplify Gen2 backends

Author: ```suggestion
    Note: Login with Username is not currently supported using Amplify Gen2 backends (Coming soon!)
```

>     Note: Login with Username is not currently supported using Amplify Gen2 backends (Coming soon!)

Reviewer: In case this doesn't happen in the immediate term:
```suggestion
    Note: Configuring sign in with `username` is not supported by Amplify Gen2 backends
```
